tests/integration/frameworks/models/__init__.py

```diff
@@ -14,6 +14,15 @@ class FrameworkTestModel:
     configurations: list[FrameworkTestModelConfiguration]
 
     save_kwargs: dict[str, t.Any] = attr.Factory(dict)
+    # when raw model method call is not simply `method(*args,
+    # **kwargs)` format or raw model method call does not simply
+    # return the outputs, then use this to override default behavior
+    # when testing raw model inputs with expected outputs
+    model_method_caller: t.Callable[
+        [FrameworkTestModel, str, list[t.Any], dict[str, t.Any]], t.Any
+    ] | None = attr.field(default=None)
+    # when framework has some special signatures requirements
+    model_signatures: dict[str, t.Any] | None = attr.field(default=None)
```

Comment:
@sauyon some changes for `FrameworkTest` for onnx or other more flexible frameworks.